tests/core/pipeline/test_pipeline_manager.py

```diff
@@ -418,13 +446,49 @@ def test_pipeline_notification_unsubscribe(mocker):
     notify_2 = notify2
 
     _PipelineManager._subscribe(callback=notify_1, pipeline=pipeline)
-    _PipelineManager._unsubscribe(notify_1, pipeline)
+    _PipelineManager._unsubscribe(callback=notify_1, pipeline=pipeline)
     _PipelineManager._subscribe(callback=notify_2, pipeline=pipeline)
     _PipelineManager._submit(pipeline.id)
 
     with pytest.raises(ValueError):
-        _PipelineManager._unsubscribe(notify_1, pipeline)
-        _PipelineManager._unsubscribe(notify_2, pipeline)
+        _PipelineManager._unsubscribe(callback=notify_1, pipeline=pipeline)
+        _PipelineManager._unsubscribe(callback=notify_2, pipeline=pipeline)
+
+
+def test_pipeline_notification_unsubscribe_multi_param():
+    Config.configure_job_executions(mode=JobConfig._DEVELOPMENT_MODE)
+    _Scheduler._update_job_config()
+
+    pipeline_config = Config.configure_pipeline(
+        "by_6",
+        [
+            Config.configure_task(
+                "mult_by_two",
+                mult_by_two,
+                [Config.configure_data_node("foo", "in_memory", Scope.PIPELINE, default_data=1)],
+                Config.configure_data_node("bar", "in_memory", Scope.PIPELINE, default_data=0),
+            )
+        ],
+    )
+
+    pipeline = _PipelineManager._get_or_create(pipeline_config)
+
+    _PipelineManager._subscribe(callback=notify_multi_param, params=["foobar", 123, 0], pipeline=pipeline)
+    _PipelineManager._subscribe(callback=notify_multi_param, params=["foobar", 123, 1], pipeline=pipeline)
+    _PipelineManager._subscribe(callback=notify_multi_param, params=["foobar", 123, 2], pipeline=pipeline)
+
+    assert len(pipeline.subscribers) == 3
+
+    pipeline.unsubscribe(notify_multi_param)
+    assert len(pipeline.subscribers) == 2
+    assert Subscriber(notify_multi_param, ["foobar", 123, 0]) not in pipeline.subscribers
+
+    pipeline.unsubscribe(notify_multi_param, ["foobar", 123, 2])
+    assert len(pipeline.subscribers) == 1
+    assert Subscriber(notify_multi_param, ["foobar", 123, 2]) not in pipeline.subscribers
+
+    with pytest.raises(ValueError):
+        pipeline.unsubscribe(notify_multi_param, ["foobar", 123, 10000])
 
```

Comment:
it would be good to submit the pipeline here and assert that the callback is called only with parameters ["foobar", 123, 1].